Automattic/wp-calypso
# Update Full Width Mobile Promo Card

Fixes https://github.com/Automattic/wp-calypso/issues/72596, please also note some of the discussions amongst developers and designers in that issue's comments.

> **Note**
> Although this PR includes ~20 changed files, the majority are relatively simple changes resulting from renaming components and moving files to different directories.

#### Proposed Changes

This PR makes some back-end and front-end tweaks to the mobile promo currently found at the bottom of the Stats page in Calypso. 

These tweaks can be grouped into the following three categories, with the purpose being to make the component easier to discover/use for developers and to bring some of the styling in alignment with other mobile promos:

<details>
 <summary>Renaming and moving components for clarity ⤵️</summary>

* The `<AppPromo>` component has been renamed to `<AppPromoSidebar  />`.
* The `<MobilePromoCard>` component has been renamed to `<AppPromoFullWidth />`.
* The [`<AnimatedIcon />`](https://github.com/Automattic/wp-calypso/tree/trunk/client/components/animated-icon) component has been moved from `client/components` to `packages/components`. This is so that it can be used with `<AppPromoFullWidth />` (which exists under the `packages` directory). To accommodate this, all of the component's existing imports and dependencies (`lottie` and `jest-canvas-mock`) have been updated to reflect its new location.
</details>

<details>
 <summary>Changes to AppPromoFullWidth component ⤵️</summary>

* The font has been updated to match the other promo cards on the Stats page. More specifically, the declaration of the font's family has been removed so that it defaults to `inherit` and the colour has been changed to `var(--studio-gray-50)`.
* The component's link has been updated from `jetpack.com/app` to `apps.wordpress.com`.
* The styling of the component's link has been slightly tweaked, to match the hover colour and underline style used for other mobile promos in Calypso.
* The QR code has been updated so that it redirects to the app stores if the user scans it with an Android or iOS device.
</details>

#### Testing Instructions

* Navigate to _My Site_ → _Stats_, scroll to the bottom of the page, and verify that the full width app promo looks as expected:
  * Scan the QR code and verify that the link directs to the play store for the device you're using.
* Verify that there are no unforeseen regressions in the sidebar app promo:
  * Navigate to the Reader in an account where you haven't dismissed the sidebar app promo. (You can manually change `shouldRenderAppPromo` to `true` [here](https://github.com/Automattic/wp-calypso/blob/dded2cf2a4d3d3770818998875bd4c3616bd93f7/client/reader/sidebar/promo.jsx#L17) if testing locally.)

#### Screenshots

**Before:**

<img width="1255" alt="Screenshot 2023-01-27 at 15 18 41" src="https://user-images.githubusercontent.com/2998162/215121557-7f8c1343-e9ee-4692-8049-476eb4a45a9a.png">

**After:**

<img width="1259" alt="Screenshot 2023-01-30 at 11 31 44" src="https://user-images.githubusercontent.com/2998162/215465531-51d2a4a0-d39b-4ed2-84a0-0b5984ff1e85.png">

#### Reverted changes 

The following changes were reverted in https://github.com/Automattic/wp-calypso/pull/72560/commits/5c2d0053260c6698896e379f5226f434f296aed0 following [this feedback](https://github.com/Automattic/wp-calypso/pull/72560#pullrequestreview-1273650573). Keeping a record here for posterity.

<details>
 <summary>Changes to AnimatedIcon component ⤵️</summary>

* An optional `playUponViewportEntry` prop has been added to allow the possibility of the animation playing each time it become visible in the browser's viewport. This is necessary for use on the Stats page, where the animated icon is only visible after scrolling to the bottom of the page (and therefore easy to miss). I opted to make this an optional prop, as it's unlikely we'll _always_ want an animation to re-play when scrolling past it. 
*  It's now also possible to pass a json object to the component's `icon` prop. Previously, it was only possible to pass the icon's path as a string, but it's more straightforward to pass a json object from the `packages` directory. 
</details>

**Video:**

https://user-images.githubusercontent.com/2998162/215121362-2642be03-9c8d-4942-95cd-107b5ced5490.mov

#### Pre-merge Checklist
